src/main.rs

```diff
@@ -10,6 +10,14 @@ fn main() {
     let mut file = File::open("./src/example.gpl").unwrap();
     let mut contents = vec![];
     file.read_to_end(&mut contents).unwrap();
+    // this removes \r as it seems to cause problems during parsing
+    for i in 0..contents.len() {
```

Comment:
There is a problem here i think..
If we have this string for example:  
```
abc\r\rabc\rabc
```
and run this code over it it will produce:
```
abc\rabcabc
```
This is because `i` goes up by 1 while the contents after the remove is now moved 1 index back  

The best way to resolve this is via a reverse loop that way `i` always matches the char you expected:
```suggestion
    for i in contents.len()..0 {
```
NOTE: i'm not sure if this is possible in rust so you'll have to test it.  
If it doesn't work you could take a look at [rust Iterators](https://doc.rust-lang.org/std/iter/trait.Iterator.html) i think they have a method for going over arrays in reverse

Reply:
I tested this, and while it did not cause an error, it failed to parse the code properly.

Before
```
   Compiling gpl v0.1.0 (C:\Users\reube\source\repos\gpl)
    Finished dev [unoptimized + debuginfo] target(s) in 0.74s
     Running `target\debug\gpl.exe`
[Function { name: Some("test"), args: [], body: Actions { list: [Variable(Variable { var_type: Let, data_type: None, name: "a", action: StaticString(String_ { content: "Hello" }) })] } }, Function { name: Some("main"), args: [], body: Actions { list: [] } }]
```

After
```
   Compiling gpl v0.1.0 (C:\Users\reube\source\repos\gpl)
    Finished dev [unoptimized + debuginfo] target(s) in 0.87s
     Running `target\debug\gpl.exe`
1: // This file is used for personal testing
2:
 ^-- Unexpected char:
3: // test function that assigns let variable 'a'
```